File: TaboolaSDKExampleObjective-C/TaboolaSDKExampleObjective-C/SDK Web/WebViewController.m
//
//  WebViewController.m
//  TaboolaSDKExampleObjective-C
//
//  Copyright © 2020 Taboola. All rights reserved.
//

#import "WebViewController.h"
#import <TaboolaSDK/TaboolaSDK.h>
#import <WebKit/WebKit.h>


@interface WebViewController () <WKNavigationDelegate, TBLWebPageDelegate>

@property (weak, nonatomic) IBOutlet UIView *webViewContainer;

@property (nonatomic, strong) WKWebView *webView;

// The TBLWebPage object that will contain the Taboola content fetched via JS
@property (nonatomic, strong) TBLWebPage* webPage;
@property (nonatomic, strong) TBLWebUnit *webUnit;

@end

@implementation WebViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _webView = [[WKWebView alloc] initWithFrame:CGRectZero];
    
    _webView.frame = self.view.frame;
    [_webViewContainer addSubview:_webView];
    
    _webPage = [[TBLWebPage alloc]initWithDelegate:self];
    _webUnit = [_webPage createUnitWithWebView:_webView];
    
    [self loadExamplePage:_webView];
}

- (void)loadExamplePage:(WKWebView*)webView {
    NSString* htmlPath = [[NSBundle mainBundle] pathForResource:@"SampleHTMLPage" ofType:@"html"];
    NSString* appHtml = [NSString stringWithContentsOfFile:htmlPath encoding:NSUTF8StringEncoding error:nil];
    [webView loadHTMLString:appHtml baseURL:[NSURL URLWithString:@"https://cdn.taboola.com/mobile-sdk/init/"]];	
}

- (void)webView:(UIView *)webView didLoadPlacementNamed:(NSString *)placementName withHeight:(CGFloat)height {
    NSLog(@"%@", placementName);
}

- (void)taboolaView:(UIView *)taboolaView didFailToLoadPlacementNamed:(NSString *)placementName withErrorMessage:(NSString *)error {
    NSLog(@"%@", error);
}

- (BOOL)onItemClick:(NSString *)placementName withItemId:(NSString *)itemId withClickUrl:(NSString *)clickUrl isOrganic:(BOOL)organic {
    // Return 'YES' for Taboola SDK to handle the click event (default behavior).
    // Return 'NO' to handle the click event yourself. (Applicable for organic content only.)
    return YES;
}

@end

File: TaboolaSDKExampleSwift/TaboolaSDKExampleSwift/SDK Web/WebViewController.swift
//
//  WebViewController.swift
//  TaboolaSDKExampleV3
//
//  Created by Liad Elidan on 22/03/2020.
//  Copyright © 2020 Liad Elidan. All rights reserved.
//

import Foundation
import TaboolaSDK

class WebViewController: UIViewController, WKNavigationDelegate, TBLWebDelegate{
    @IBOutlet weak var webViewContainer: UIView!
    var webView = WKWebView()

    override func viewDidLoad() {
        super.viewDidLoad()
        
        webView.frame = view.frame
        webViewContainer.addSubview(webView)
        
        let jsPage =  TBLWebPage.init(delegate: self)
        jsPage.createUnit(with: webView)
                
        try? loadExamplePage()
    }
    
    func loadExamplePage() throws {
        guard let htmlPath = Bundle.main.path(forResource: "SampleHTMLPage", ofType: "html") else {
            print("Error loading HTML")
            return
        }
        let appHtml = try String.init(contentsOfFile: htmlPath, encoding: .utf8)
        webView.loadHTMLString(appHtml, baseURL: URL(string: "https://cdn.taboola.com/mobile-sdk/init/"))
    }
    
    func webView(_ webView: UIView!, didLoadPlacementNamed placementName: String!, withHeight height: CGFloat) {
        print("Placement name: \(String(describing: placementName)) has been loaded with height: \(height)")
    }
    
    func webView(_ webView: WKWebView, didFail navigation: WKNavigation!, withError error: Error) {
        print(error as Any)
    }
    
    func onItemClick(_ placementName: String!, withItemId itemId: String!, withClickUrl clickUrl: String!, isOrganic organic: Bool) -> Bool {
        if (!organic) {
            return false;
        }
        return true;
    }
}
